### PR TITLE
folder check at dp clone

### PR DESCRIPTION
**Pull request by sbohora**

### CODE REVIEW

**Review by camorosi — COMMENTED** [review]
The folder additions to dp_clone look good to me!

The other checks added to dp_init aren't strictly relevant to this issue (because if you're running dp_clone you wouldn't need to run dp_init) but seem like helpful additions regardless.

File: R/dp_init.R [hunk 1]
```diff
@@ -77,6 +77,14 @@ dp_init <- function(project_path = fs::path_wd(),
   commit_description <- "dp init"
   wd0 <- fs::path_wd()
 
+  if(fs::dir_exists(project_path)){
```
Comment (amashadihossein) [inline R/dp_init.R:80]: Addition of this leads to confusion on what the desired behavior should be based on line 88:
- line 80 says -> stop if dir doesn't exist
- line 88 says -> create if dir doesn't exist
One should be chosen and the other deleted.  I suggest we remove line 80 as the issue it is try to address is addressed by line 84 `is_valid_dp_repository(project_path)`

Comment (amashadihossein) [reply R/dp_init.R:80]: note `is_valid_dp_repository(project_path)` checks all `all` elements of the repository ("git" & "dp" & "renv" & "branch"). We may want an `or` logic so that if any of those elements are passed it returns TRUE and hence stops

Comment (sbohora) [reply R/dp_init.R]: @amashadihossein 

I made the changes to address the issues. `is_valid_dp_repository` has been revised to check for all ("git" & "dp" & "renv" & "branch") versus any checks. Please let me know if this seems alright or we may want to seek a way around it. Thanks.

Comment (camorosi) [reply R/dp_init.R]: The error message you have on line 81 of dp_init.R specifically references a dp repository. So if the user is in git or renv repository (but not a dp repository) they will get the error message "You are already inside a dp repository". Recommend changing the error message to something more specific to which of the checked elements already exist. There's a verbose option on `is_valid_dp_repository` that could probably be used here.

File: R/dp_init.R [hunk 2]
```diff
@@ -77,15 +77,12 @@ dp_init <- function(project_path = fs::path_wd(),
   commit_description <- "dp init"
   wd0 <- fs::path_wd()
 
-  if (is_valid_dp_repository(project_path, check_any_checks = T)){
+  if (is_dp_initiated(project_path)){
```
[the remark this answers is not in the document]
Comment (camorosi) [reply R/dp_init.R]: I think it should be whatever the parent directory of project_path is, so I changed it to `path = fs::path_norm(fs::path(project_path, ".."))` Let me know if this look ok

File: R/dp_init.R [hunk 3]
```diff
@@ -77,10 +77,13 @@ dp_init <- function(project_path = fs::path_wd(),
   commit_description <- "dp init"
   wd0 <- fs::path_wd()
 
+  if (is_dp_initiated(path = fs::path_norm(fs::path(project_path, "..")), checks = c("git", "renv", "dp"))){
```
Comment (amashadihossein) [inline R/dp_init.R:80]: The definition of `is_dp_initiated` does not take `checks` as a parameter. If so, checks need to be removed. Does this not throw an error?

File: R/dx.R [hunk 4]
```diff
@@ -1,47 +1,66 @@
...
+
   dp_repository <- all(sapply(dx,isTRUE))
   return(dp_repository)
-  
 }
 
+#' @title Determine if dp repository initiated
+#' @description Looks at the path, runs `dp_repository_check` and returns TRUE
+#' if any TRUE
+#' @details All diagnostic tests to check validity of dp repository are run
```
Comment (amashadihossein) [inline R/dx.R:46]: This seems irrelevant as `is_dp_initiated` doesn't take any parameters other than `path` and `verbose`. Please correct the comment
